modules/internal/DefaultAssociative.chpl

```diff
@@ -566,14 +566,15 @@ module DefaultAssociative {
 
     // ref version
     proc dsiAccess(idx : idxType, haveLock = false) ref {
-      const shouldLock = dom.parSafe && !haveLock;
+      const arrOwnsDom = dom._arrs.length == 1;
+      const shouldLock = dom.parSafe && !haveLock && arrOwnsDom;
```

Comment:
Right, but what worries me is that if a user didn't know that we supported the example you wrote, if they wrote the longer version of the same, it would be wrong. 
``` chapel
forall key in collection {
  if D.member(key) then
    A[key] += 1;
  else
   D += 1;
```

which makes me think that the 1:1 auto-domain addition is not a shorthand at all.